test/e2e/framework/kcp.go

```diff
@@ -256,15 +256,21 @@ func (c *kcpServer) Run(opts ...RunOption) error {
 		return err
 	}
 
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
 	c.t.Cleanup(func() {
 		// Ensure child process is killed on cleanup
 		err := cmd.Process.Kill()
```

Comment:
I was fiddling around with this last night. I couldn't get the kill to actually kill the process if I wanted the goroutine at L272 to actually do anything (I added channel coordination to wait for the log filtering to finish before completely exiting this Run func)

Reply:
Have to look deeper into this again. What I basically want in CI is to block when we don't shutdown in time. I.e. I want that we notice (might be painful) when we regress.

Reply:
I'm sure one of us can get to the bottom of it. I was definitely seeing the log filtering goroutine never having the defer call executed